src/resolve/resolved-dns-transaction.c

```diff
@@ -1262,9 +1266,9 @@ static int dns_transaction_prepare(DnsTransaction *t, usec_t ts) {
                 }
         }
 
-        /* Check the cache, but only if this transaction is not used
+        /* Check the cache, but only if enabled and this transaction is not used
          * for probing or verifying a zone item. */
-        if (set_isempty(t->notify_zone_items)) {
+        if (t->scope->manager->enable_cache && set_isempty(t->notify_zone_items)) {
```

Comment:
I know it's not necessary, I just found it to be more symmetric, and avoids querying the cache if we know it isn't being used. But if that's cheap, I'm happy to drop this bit.